src/linux/functions.cpp

```diff
@@ -15,27 +15,70 @@
  */
 
 #include <sys/mount.h>
+#include <mntent.h>
 #include <errno.h>
 #include "../mountutils.hpp"
 
 NAN_METHOD(UnmountDisk) {
   if (!info[1]->IsFunction()) {
-    return Nan::ThrowError("Callback must be a function");
+    return Nan::ThrowTypeError("Callback must be a function");
   }
 
   v8::Local<v8::Function> callback = info[1].As<v8::Function>();
 
   if (!info[0]->IsString()) {
-    return Nan::ThrowError("Device argument must be a string");
+    return Nan::ThrowTypeError("Device argument must be a string");
   }
 
   v8::String::Utf8Value device(info[0]->ToString());
-  int code = umount(reinterpret_cast<char *>(*device));
 
-  if (code == 0) {
-    YIELD_NOTHING(callback);
+  const char *device_path = reinterpret_cast<char *>(*device);
+  const char *mount_path = NULL;
+
+  // Get mountpaths from the device path, as `umount(device)`
+  // has been removed in Linux 2.3+
+  struct mntent *mount_entity;
+  FILE *proc_mounts;
+
+  proc_mounts = setmntent("/proc/mounts", "r");
+
+  if (proc_mounts == NULL) {
+    v8::Local<v8::Value> argv[1] = {
+      Nan::ErrnoException(errno, "setmntent", NULL, "/proc/mounts")
+    };
+    Nan::MakeCallback(Nan::GetCurrentContext()->Global(), callback, 1, argv);
+    return;
+  }
+
+  // TODO(jhermsmeier): Support únmounting multiple mountpoints (?)
```

Comment:
Oh, well I guess I must have subtly screwed up my keyboard mapping on the Linux machine - Fixed now